packages/flutter/test/material/scaffold_test.dart

```diff
@@ -1596,6 +1596,142 @@ void main() {
     expect(scaffoldState.isDrawerOpen, true);
   });
 
+  testWidgets('Drawer does not open or close with drag gesture when it is disabled', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          drawer: const Drawer(
+            child: Text('Drawer'),
+          ),
+          drawerEnableDragGesture: true,
+          body: const Text('Scaffold Body'),
+          appBar: AppBar(
+            centerTitle: true,
+            title: const Text('Title'),
+          ),
+        ),
+      ),
+    );
+    ScaffoldState scaffoldState = tester.state(find.byType(Scaffold));
+    expect(scaffoldState.isDrawerOpen, false);
+
+    // Test that we can open and close drawer with drag gesture when
+    // `Scaffold.drawerEnableDragGesture` is true.
+    await tester.dragFrom(const Offset(0, 100), const Offset(300, 0));
+    await tester.pumpAndSettle();
+    expect(scaffoldState.isDrawerOpen, true);
+
+    await tester.dragFrom(const Offset(300, 100), const Offset(-300, 0));
+    await tester.pumpAndSettle();
+    expect(scaffoldState.isDrawerOpen, false);
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          drawer: const Drawer(
+            child: Text('Drawer'),
+          ),
+          drawerEnableDragGesture: false,
+          body: const Text('Scaffold body'),
+          appBar: AppBar(
+            centerTitle: true,
+            title: const Text('Title'),
+          ),
+        ),
+      ),
+    );
+    scaffoldState = tester.state(find.byType(Scaffold));
+    expect(scaffoldState.isDrawerOpen, false);
+
+    // Test that we cannot open drawer with drag gesture when
+    // `Scaffold.drawerEnableDragGesture` is false.
+    await tester.dragFrom(const Offset(0, 100), const Offset(300, 0));
+    await tester.pumpAndSettle();
+    expect(scaffoldState.isDrawerOpen, false);
+
+    // Test that we cannot close drawer with drag gesture when
```

Comment:
I came to the realization that users may actually want the close drag gesture to still close the drawer. From what it sounded like in https://github.com/flutter/flutter/issues/31127, they run into problems when users accidentally scroll open the drawer while trying to scroll through other UI like lists. 

I'll post a follow up question in the original GitHub issue and see what they say, since forcing the close to be programmatic might be counter to what was actually intended.

Reply:
Makes sense.

Also you can also tap outside of the drawer to close it. Just not drag and close it.

Reply:
I changed it now so that we can now close the drawer with a drag gesture, after comments in the ticket.